server/src/diskdb.ts

```diff
@@ -0,0 +1,85 @@
+import { promises as fs } from 'fs';
+import {
+  Result,
+  Database,
+  DatabaseType,
+  DatabaseModel,
+} from '../../frontend/src/types';
+import * as uuid from 'uuid';
+
+export class Disk implements Database {
+  public type = DatabaseType.Disk;
+  db: Map<string, DatabaseModel>;
+  path = './db.json';
+
+  async saveDB() {
+    const serialiseDB = JSON.stringify(Array.from(this.db.entries()));
```

Comment:
You'll need a lock (or some other way of synchronisation) around this to make sure there's not more than one concurrent call. Otherwise you may end up with inconsistent results on disk

Reply:
Would using writeFileSync be enough to prevent against concurrent calls? Otherwise i can use https://www.npmjs.com/package/async-mutex or the code from https://spin.atomicobject.com/2018/09/10/javascript-concurrency/. Thoughts?

Reply:
Have implemented the writeFileSync for now. But can pivot to the other solutions if i'm missing something.

Reply:
`writeFileSync` won't solve the locking problem.  I believe what @zsims was trying to say was that if another use happened to be working on the same board at the same time, you might end up overwriting each other's data without releasing it. When you receive a lock, the application is basically saying to you, "as long as you have that lock, you're the only one able to mutate the file/DB/etc". 

You can read more about it on [here](https://en.wikipedia.org/wiki/File_locking)

A quick Google revealed [this library](https://github.com/moxystudio/node-proper-lockfile) that might help facilitate the lock process

Reply:
Implemented the lock using that lib, thanks for the suggestion.